server/sonar-web/src/main/js/apps/custom-metrics/components/Form.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { Metric } from '../../../app/types';
+import DeferredSpinner from '../../../components/common/DeferredSpinner';
+import SimpleModal from '../../../components/controls/SimpleModal';
+import { translate } from '../../../helpers/l10n';
+import Select, { Creatable } from '../../../components/controls/Select';
+
+export interface MetricProps {
+  description: string;
+  domain?: string;
+  key: string;
+  name: string;
+  type: string;
+}
+
+interface Props {
+  confirmButtonText: string;
+  domains: string[];
+  metric?: Metric;
+  header: string;
+  onClose: () => void;
+  onSubmit: (data: MetricProps) => Promise<void>;
+  types: string[];
+}
+
+interface State extends MetricProps {}
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: An interface declaring no members is equivalent to its supertype. ([typescript:S4023](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4023))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fcustom-metrics&issues=AWF1nUAi0orRW2HVTuy_&open=AWF1nUAi0orRW2HVTuy_)

[ISSUE_KEY]: AWF1nUAi0orRW2HVTuy_